src/math/vec2.h

```diff
@@ -0,0 +1,95 @@
+//
+// Created by ajer on 16.12.18.
+//
+
+#pragma once
+#include "pch.h"
+#include <cmath>
+
+namespace Bald::Math{
+
+    class Vec2{
+    public:
+        /*
+         * @fn Vec2
+         * @brief constructor
+         * @param [int] x
+         * @param [int] y
+        */
+        Vec2(double x = 0.0, double y = 0.0)
+                :m_x(x), m_y(y){}
+        /*
+         * @fn Vec2
+         * @brief copy constructor
+         * @param [const Vec2&] other vector
+        */
+        Vec2(const Vec2& other)
+                :m_x(other.m_x), m_y(other.m_y){}
+
+
+        /*
+         * @fn GetX
+         * @brief returns member X variable
+        */
+        [[nodiscard]] constexpr inline double GetX()const noexcept{return m_x;}
+        /*
+         * @fn GetY
+         * @brief returns member Y variable
+        */
+        [[nodiscard]] constexpr inline double GetY()const noexcept{return m_y;}
+
+        /*
+         * @fn Print
+         * @brief prints vec
+        */
+        inline void Print()const  {
```

Comment:
we should overload operator "<<" it would be more handy

Reply:
done